cluster-api/cloud/google/machineactuator.go

```diff
@@ -653,24 +679,22 @@ func (gce *GCEClient) handleMachineError(machine *clusterv1.Machine, err *apierr
 	return err
 }
 
-func (gce *GCEClient) getImage(machine *clusterv1.Machine, config *gceconfig.GCEProviderConfig) (image string, isPreloaded bool) {
+func (gce *GCEClient) getImagePath(img string, project string) (imagePath string) {
 	defaultImg := "projects/ubuntu-os-cloud/global/images/family/ubuntu-1710"
-	project := config.Project
-	img := config.Image
 
 	// A full image path must match the regex format. If it doesn't, we'll assume it's just the image name and try to get it.
 	// If that doesn't work, we will fall back to a default base image.
 	matches := regexp.MustCompile("projects/(.+)/global/images/(family/)*(.+)").FindStringSubmatch(img)
 	if matches == nil {
-		// Only the image name was specified in config, so check if it is preloaded in the project specified in config.
+		// Only the image name was specified in config, so check if it exists in the project specified in config.
```

Comment:
I think we can say that image path needs to be fully specified in the config now. And we can drop the project parameter as it's only needed for when the full path wasn't specified.